wdl-doc/src/lib.rs

```diff
@@ -67,282 +170,198 @@ impl Document {
+/// `docs` directory within the workspace.
+pub async fn document_workspace(
+    workspace: impl AsRef<Path>,
+    stylesheet: Option<impl AsRef<Path>>,
+    overwrite: bool,
+) -> Result<PathBuf> {
+    let workspace_abs_path = absolute(workspace)?;
+    let stylesheet = stylesheet.and_then(|p| absolute(p.as_ref()).ok());
+
+    if !workspace_abs_path.is_dir() {
+        return Err(anyhow!("Workspace is not a directory"));
     }
 
-    let abs_path = std::path::absolute(&path)?;
-
-    let docs_dir = abs_path.clone().join(DOCS_DIR);
+    let docs_dir = workspace_abs_path.join(DOCS_DIR);
+    if overwrite && docs_dir.exists() {
+        std::fs::remove_dir_all(&docs_dir)?;
+    }
     if !docs_dir.exists() {
         std::fs::create_dir(&docs_dir)?;
     }
 
     let analyzer = Analyzer::new(DiagnosticsConfig::new(rules()), |_: (), _, _, _| async {});
-    analyzer.add_directory(abs_path.clone()).await?;
+    analyzer.add_directory(workspace_abs_path.clone()).await?;
     let results = analyzer.analyze(()).await?;
 
+    let mut docs_tree = if let Some(ss) = stylesheet {
+        docs_tree::DocsTree::new_with_stylesheet(docs_dir.clone(), ss)?
+    } else {
+        docs_tree::DocsTree::new(docs_dir.clone())
+    };
+
     for result in results {
-        let cur_path = result
-            .document()
-            .uri()
-            .to_file_path()
-            .expect("URI should have a file path");
-        let relative_path = match cur_path.strip_prefix(&abs_path) {
-            Ok(path) => path,
-            Err(_) => &PathBuf::from("external").join(cur_path.strip_prefix("/").unwrap()),
+        let uri = result.document().uri();
+        let rel_wdl_path = match uri.to_file_path() {
+            Ok(path) => match path.strip_prefix(&workspace_abs_path) {
+                Ok(path) => path.to_path_buf(),
+                Err(_) => {
+                    PathBuf::from("external").join(path.components().skip(1).collect::<PathBuf>())
+                }
+            },
+            Err(_) => PathBuf::from("external").join(
+                uri.path()
+                    .strip_prefix("/")
+                    .expect("URI path should start with /"),
+            ),
         };
-        let cur_dir = docs_dir.join(relative_path.with_extension(""));
+        let cur_dir = docs_dir.join(rel_wdl_path.with_extension(""));
         if !cur_dir.exists() {
             std::fs::create_dir_all(&cur_dir)?;
         }
-        let name = cur_dir
-            .file_name()
-            .expect("current directory should have a file name")
-            .to_string_lossy();
         let ast_doc = result.document().node();
         let version = ast_doc
             .version_statement()
-            .expect("document should have a version statement")
-            .version();
-        let preamble = fetch_preamble_comments(ast_doc.clone());
+            .expect("Document should have a version statement");
```

Comment:
```suggestion
            .expect("document should have a version statement");
```

These expect statements should be all lowercase to follow the existing convention (unless you're starting with a proper noun, like `WDL`).

Reply:
Please double check all the expect statements here.